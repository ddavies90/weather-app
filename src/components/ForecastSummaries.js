import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastSummaries.css";
import ForecastSummary from "./ForecastSummary";

const ForecastSummaries = ({ forecasts, onForecastSelect }) => {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => {
        const { dt: date, temp, weather, isSelected } = forecast;
        const { id: icon, main: description } = weather[0];
        return (
          <ForecastSummary
            key={date}
            date={date}
            icon={icon.toString()}
            temperature={temp}
            description={description}
            onSelect={onForecastSelect}
            isSelected={isSelected}
          />
        );
      })}
    </div>
  );
};

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      dt: PropTypes.number.isRequired,
      temp: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number
      }),
      weather: PropTypes.arrayOf(
        PropTypes.shape({
          main: PropTypes.string.isRequired,
          id: PropTypes.string.isRequired
        }).isRequired
      ).isRequired
    })
  ).isRequired,
  onForecastSelect: PropTypes.func.isRequired
};

export default ForecastSummaries;
