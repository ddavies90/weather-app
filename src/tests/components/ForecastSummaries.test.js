import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("<ForecastSummaries />", () => {
  const validProps = [
    {
      id: 1,
      date: 1525046400000,
      temperature: {
        max: 11,
        min: 4
      },
      description: "Clear",
      icon: "800"
    },
    {
      id: 2,
      date: 1525132800000,
      temperature: {
        max: 13,
        min: 8
      },
      description: "Stormy",
      icon: "211"
    },
    {
      id: 3,
      date: 1525219200000,
      temperature: {
        max: 1,
        min: -2
      },
      description: "Heavy Snow",
      icon: "602"
    }
  ];

  const handleForecastSelect = jest.fn();

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps}
        onForecastSelect={handleForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays all forecast summary components", () => {
    render(
      <ForecastSummaries
        forecasts={validProps}
        onForecastSelect={handleForecastSelect}
      />
    );

    expect(screen.getAllByTestId("forecast-summary")).toHaveLength(3);
  });
});
