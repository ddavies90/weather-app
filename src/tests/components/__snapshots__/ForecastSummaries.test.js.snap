// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastSummaries /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Mon, 30 Apr
      </div>
      <i
        class="wi wi-day-sunny"
        data-testid="forecast-icon"
        name="owm"
      />
      <div
        class="forecast-summary__temperature"
      >
        11°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Clear
      </div>
      <button
        class="forecast-summary__btn"
        data-testid="button-click"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Tue, 1 May
      </div>
      <i
        class="wi wi-day-thunderstorm"
        data-testid="forecast-icon"
        name="owm"
      />
      <div
        class="forecast-summary__temperature"
      >
        13°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stormy
      </div>
      <button
        class="forecast-summary__btn"
        data-testid="button-click"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Wed, 2 May
      </div>
      <i
        class="wi wi-day-snow"
        data-testid="forecast-icon"
        name="owm"
      />
      <div
        class="forecast-summary__temperature"
      >
        1°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Heavy Snow
      </div>
      <button
        class="forecast-summary__btn"
        data-testid="button-click"
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
